### DISCUSSION TITLE
How to skip the build in Concourse whenever there are no new commits/changes.

### DISCUSSION THREAD

**Discussion by Keets25** [concourse/concourse#7527] [Help & Support]
I have a single concourse pipeline where I build three docker images. Of all the three images, Two images doesn't have regular updates/changes and the changes occur occasionally. So I wanted to skip the build part in concourse for this two images. Trigger value is set to true. But when I trigger the build manually, it is actually not skipping the build though there are no latest changes/commits available. Can I get some help on how to resolve this based on certain conditions. I also read the document that says, concourse doesn't have built in support for conditionals and loops.

[test-concouse-skip-build.docx](https://github.com/concourse/concourse/files/7126870/test-concouse-skip-build.docx)

**Comment by x3nb63** [comment]
Why do you trigger it?

Given the above I would separate the two images into another job _(or even one job for each)_, which are triggered by their resource, respectively.

Then your third image would be yet another job, which can trigger independently.

**Reply by Keets25** [reply]
I have security scanning tools like white-source configured to our pipeline. So incase of issues with it I might need to trigger other jobs manually at times but not necessarily. 
I have created 4 individual jobs . Job A,B,C,D . All this three jobs, B,C,D are dependent on A but have independent resources. I have included serial_groups but just wanted to know will serial_groups stops triggering other jobs if Job-A is failed. I don't want other jobs to be triggered when Job-A fails. Does serial_groups be helpful in that scenario or is there any better way to do it? .. I was not able to use "passed" boolean value since the resources are independent of each other.

**Reply by x3nb63** [reply]
from top of my head:

a _serial_group_ does not do any skipping; it only blocks in time to ensure only one of the jobs listed runs concurrently. Should not be useful for this problem.

I am not sure there is a feature to do grouping of the triggers on job level, to say: _"run this job only if *all* these triggers have a new version"_?

there are a number of "workflow-altering" resources that may be suitable to describe you problem using `resources:` definitions. One is `gate-resource`, another is `pool-resource` maybe even `semver-resource` could be used in such a way, to achieve a _"fan in"_ job flow, perhaps. Check https://resource-types.concourse-ci.org/ on what there is.

(I also remember the old discuss forum had ideas for job-flow logic)
